app/javascript/utils/lazy-highlight-all.ts

```diff
@@ -0,0 +1,49 @@
+export function lazyHighlightAll(): void {
+  let highlighted = false
+
+  function checkAndHighlight(): void {
+    if (highlighted) return
+
+    // only load highlightjs when a code block with lang-... or language-... classname exists in the DOM
+    if (
+      document.querySelector('code[class^=lang-]') !== null ||
+      document.querySelector('code[class^=language-]') !== null
+    ) {
```

Comment:
Maybe move this logic to a separate function? I would probably also remove the logic here and put it around the `checkAndHighlight()` that handles the initial highlighting.